internal/commands/build/build.go

```diff
@@ -0,0 +1,240 @@
+package build
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/deislabs/cnab-go/bundle"
+	cnab "github.com/deislabs/cnab-go/driver"
+	"github.com/docker/app/internal/packager"
+	"github.com/docker/app/types"
+	"github.com/docker/buildx/build"
+	"github.com/docker/buildx/driver"
+	_ "github.com/docker/buildx/driver/docker" // required to get default driver registered, see driver/docker/factory.go:14
+	"github.com/docker/buildx/util/progress"
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/distribution/reference"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/auth/authprovider"
+	"github.com/moby/buildkit/util/appcontext"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type buildOptions struct {
+	noCache  bool
+	progress string
+	pull     bool
+	tag      string
+	out      string
+}
+
+func Cmd(dockerCli command.Cli) *cobra.Command {
+	var opts buildOptions
+	cmd := &cobra.Command{
+		Use:     "build [APPLICATION]",
+		Short:   "Build service images for the application",
+		Example: `$ docker app build myapp.dockerapp`,
+		Args:    cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			tag, err := runBuild(dockerCli, args[0], opts)
+			if err == nil {
+				fmt.Printf("Successfully build %s\n", tag.String())
+			}
+			return err
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.noCache, "no-cache", false, "Do not use cache when building the image")
+	flags.StringVar(&opts.progress, "progress", "auto", "Set type of progress output (auto, plain, tty). Use plain to show container output")
+	flags.BoolVar(&opts.pull, "pull", false, "Always attempt to pull a newer version of the image")
+	flags.StringVarP(&opts.out, "output", "o", "", "Dump generated bundle into a file")
```

Comment:
I wonder if this flag is really useful for the user 🤔 I think it exposes the bundle concept to the user, but everywhere else we are talking about building an app image. I guess we use it for e2e testing, but maybe we can find the bundle directly into the bundlestore instead?

Reply:
indeed. Quick fix could be to hide this option, otherwise need to give e2e tests a way to retrieve a bundle file.

Reply:
I think you can in e2e tests, each time we create an [e2e docker command](https://github.com/docker/app/blob/master/e2e/main_test.go#L40) it has its own docker config directory, so each test will have its own bundle store. That said, the temporary config dir is not currently returned, we may have to update that part.
Does that make sense?